asv/commands/profiling.py

```diff
@@ -166,7 +166,7 @@ def run(cls, conf, benchmark, revision=None, gui=None, output=None,
 
             benchmarks = Benchmarks.discover(conf, repo, environments,
                                              [commit_hash],
-                                             regex='^{0}$'.format(benchmark))
+                                             regex='^{0}\('.format(benchmark))
```

Comment:
But does it now match to non-parametrized benchmarks any more?

Reply:
With current implementation it will match because we regex search against `benchmark_name()`.
But I think we should keep matching against `benchmark_name` for non-parameterized benchmarks (at least for backward compatibility).
For parameterized benchmark `asv profile` will require a fully qualified benchmark name with param selection. I'll update the `--help` command to reflect this.